### PR TITLE
Navigation Bar: Redux refactor

### PR DESCRIPTION
**Pull request by Copons**
First step of the complete `Navigation Bar` and `TagList` refactor.

In this PR we aim to move props and actions away from `app.jsx` and instead `connect` them to the components.

### CODE REVIEW

File: lib/navigation-bar.jsx [hunk 1]
```diff
@@ -20,7 +29,16 @@ export default React.createClass( {
 	],
 
 	handleClickOutside: function() {
-		this.props.onOutsideClick( true );
+		//this.props.onOutsideClick( true );
```
Comment (dmsnell) [inline lib/navigation-bar.jsx]: commented code?

remove much?

**Comment by Copons** [comment]
@dmsnell removed, sorry about that 😭

**Comment by dmsnell** [comment]
:shipit: if you have tested each of the functions which were moved

**Comment by Copons** [comment]
@dmsnell The functions were tested and working.
Just, there was one last commit taking care of a couple of regressions that for some reasons I didn't push yesterday.

**Comment by dmsnell** [comment]
:shipit:

**Pull request merged**